sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/VisualStudioCoverageXmlReportParserTest.java

```diff
@@ -142,19 +139,18 @@ public void valid_with_multiple_getter_setter_per_line() throws Exception {
 
     String filePath = new File("GetSet\\Bar.cs").getCanonicalPath();
 
-    assertThat(coverage.files()).contains(
+    assertThat(coverage.files()).containsOnly(
       filePath,
       new File("GetSetTests\\BarTests.cs").getCanonicalPath()
     );
 
     assertThat(coverage.hits(filePath))
       .hasSize(1)
-      .contains(
-        Assertions.entry(11, 2));
+      .containsOnly(Assertions.entry(11, 2));
 
-    List<BranchCoverage> branchCoverages = coverage.getBranchCoverage(filePath);
-    assertThat(branchCoverages).hasSize(1);
-    assertThat(branchCoverages.get(0)).isEqualTo(new BranchCoverage(11, 6, 2));
+    assertThat(coverage.getBranchCoverage(filePath))
+      .hasSize(1)
```

Comment:
I think that we don't need `hasSize` if we call `containsExactly`

Reply:
indeed, for small collections.

for big collections (or collections with more than 1 item), if it fails it can mean:
- either the number is different
- or one element is different

and then the dev must do a manual diff between the outputted contents of the expected and actual collection, which is painful

Reply:
I only removed `hasSize(1)`, as I said for more than one element failing due to size is helpful